### PR TITLE
Massive workflow refactor: split into reusable parts

### PR DESCRIPTION
**Pull request by fernflower**
3 separate workflows have been introduced - reuse-copr-build,
reust-tests-7to8 and reuse-tests-8to9.
This should make adding new tests a bit easier.

### CODE REVIEW

**Comment by fernflower** [comment]
Proof of work:
/rerun - https://github.com/fernflower/leapp/actions/runs/2447483469
/rurun-all - https://github.com/fernflower/leapp/actions/runs/2447653500

**Comment by pirat89** [comment]
@fernflower review in progress. pls update the `pr-welcome-msg.yml` as well

File: .github/workflows/reuse-tests-8to9.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+name: reuse-tests-7to8@TF
```
Comment (pirat89) [inline .github/workflows/reuse-tests-8to9.yml:1]:
```suggestion
name: reuse-tests-8to9@TF
```

File: .github/workflows/reuse-tests-8to9.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        required: True
+      TF_API_KEY:
+        required: True
+    inputs:
+      copr_artifacts:
+        required: True
+        type: string
+      tmt_plan_regex:
+        type: string
+        default: "^(?!.*c2r)(?!.*sap)(?!.*7to8)"
```
[the remark this answers is not in the document]
Comment (fernflower) [reply .github/workflows/reuse-tests-8to9.yml:16]: No, ?! means exclusion. My regex is correct, it excludes 7to8 tests for 8to9 upgrade.

File: .github/workflows/reuse-tests-8to9.yml (hunk 1 above)
Comment (fernflower) [reply .github/workflows/reuse-tests-8to9.yml:1]: Nice catch, thanks!

File: .github/workflows/reuse-tests-8to9.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pirat89) [reply .github/workflows/reuse-tests-8to9.yml:16]: ah, sorry. wrong reading. thanks

**Comment by fernflower** [comment]
Okay, final run:
/rerun - https://github.com/fernflower/leapp/actions/runs/2454018337
/rerun-all 901 - https://github.com/fernflower/leapp/actions/runs/2454019736

**Comment by fernflower** [comment]
Okay, this https://github.com/oamg/leapp/pull/776/commits/7cf2eeea4c0c3f30a7e8e92e196574e8a368f51d should fix it.
Removing WIP label, scheduling all runs now.
/rerun-all 901 - https://github.com/fernflower/leapp/actions/runs/2454223657
/rerun 900 - https://github.com/fernflower/leapp/actions/runs/2454225147
/rerun - https://github.com/fernflower/leapp/actions/runs/2454225713
/rerun-all - https://github.com/fernflower/leapp/actions/runs/2454227848

**Comment by fernflower** [comment]
Okay, another try with https://github.com/oamg/leapp/pull/776/commits/b67c84b037fd2ee48951e0739f4dd8a493ac5140 =)
/rerun-all 901 - https://github.com/fernflower/leapp/actions/runs/2454290013
/rerun 900 - https://github.com/fernflower/leapp/actions/runs/2454291311
/rerun-all - https://github.com/fernflower/leapp/actions/runs/2454291735
/rerun - https://github.com/fernflower/leapp/actions/runs/2454292124

**Review by pirat89 — APPROVED** [review]

**Pull request merged**